tests/unit/sys/test_msg_stack.cxx

```diff
@@ -29,6 +29,61 @@ TEST(MsgStackTest, PopTest) {
   EXPECT_EQ(dump, expected_dump);
 }
 
+TEST(MsgStackTest, PopValueTest) {
+  MsgStack msg_stack;
+
+  msg_stack.push("First");
+  msg_stack.push("Second");
+  msg_stack.push("Third");
+  msg_stack.pop(2);
+
+  auto dump = msg_stack.getDump();
+  auto first_dump = "====== Back trace ======\n -> Second\n -> First\n";
+  EXPECT_EQ(dump, first_dump);
+
+  msg_stack.pop(-5); // Points to first message only
+  dump = msg_stack.getDump();
+  auto second_dump = "====== Back trace ======\n";
+  EXPECT_EQ(dump, second_dump);
+}
+
+TEST(MsgStackTest, ReallocStorageTest) {
+  MsgStack msg_stack;
+
+  for (int i = 0; i < 20; i++) {
+    msg_stack.push("Message %i", i);
```

Comment:
Should this `EXPECT_NO_THROW`? Or possibly have a check that there are the correct number of messages in the `msg_stack` at the end?

Reply:
There are no throws in `push` so there's no way this could throw at the moment. We could check the number of messages but this is independent of the path through `push` so should probably be a standalone test if we want to test that bit I guess (and is essentially in `PushReturnTest`)

Reply:
But yes I see the issue that there is no `EXPECT...` at the moment in this test.

I'd guess the best thing, if a bit ugly would be to check the contents of the back-trace.

Reply:
Updated to actually test something now.